### ISSUE TITLE
Why are there no interfaces in the dart?

### ISSUE THREAD

**Issue by swedesjs** [dart-lang/sdk#47172]
Hello everyone, I would like to know why there are still no interfaces in the dart for describing the key and its contents?

**Comment by lrhn** [comment]
I'm not sure what "key" you're asking about.
Dart classes define interfaces, so there are interfaces in the language.

**Comment by swedesjs** [comment]
> Я не уверен, о каком «ключе» вы спрашиваете.
> Классы Dart определяют интерфейсы, поэтому в языке есть интерфейсы.

I mean interfaces for Map types

**Comment by lrhn** [comment]
Sounds like https://stackoverflow.com/questions/69131599/how-to-define-the-shape-of-a-map-type-in-dart

Dart has objects and classes, and sound static type checking, unlike something like JavaScript. 
You are not intended to use `Map`s for heterogenous structured data, you just should define a class, and then you get an interface.
Dart maps are mutable, and the map interface can be implemented in many different ways (`SplayTreeMap`, `LinkedHashMap`, etc.) There is no way to ensure and check the *structure* of such a map statically.

In short, because Dart is not JavaScript.
